golos.referral/src/golos.referral.cpp

```diff
@@ -61,6 +63,23 @@ void referral::addreferral(name referrer, name referral, uint32_t percent,
     });
 }
 
+void referral::transfer(name referral, asset quantity) {
```

Comment:
The user should be able to self-pay off the referral program. Therefore, calling this method is not appropriate (as it requires the rights of the owner of this smart contract).
In such cases, in other places, we catch the notification of the transfer of funds to the account of the contract and perform actions. See: https://github.com/GolosChain/golos-smart/blob/master/golos.vesting/golos.vesting.cpp#L20-L21